src/modules/PlayerClassStatsModule.ts

```diff
@@ -211,12 +220,12 @@ class PlayerClassStatsModule implements events.IStats {
     }
 
     finish() {
-        const self = this;
+        const self = this
         //calculate average damage done on finish
-        this.players.forEach(function(playerStats,key){
-            playerStats.forEach(function(classStarts,key){
-                classStarts.weapons.forEach(function(weaponStats,key){
-                    weaponStats.avgDamage = self.getMean(weaponStats.avgDamages);
+        this.players.forEach(function (playerStats, key) {
```

Comment:
replace the `function()` keyword with arrow functions to preserve `this` scope to the parent.